_data/import-public.rb

```diff
@@ -12,6 +12,23 @@
 require 'hash-joiner'
 require 'safe_yaml'
 
+PROJECT_FILES = Dir["private/projects/*.yml"]
+
+private_projects_yaml_file = "private/projects.yml"
+project_data = SafeYAML.load_file(private_projects_yaml_file, :safe=>true)
+PROJECT_FILES.each do |source|
+  data = SafeYAML.load_file(source, :safe=>true)
+  unless data
+    puts "Failed to parse #{source}"
+    exit 1
+  end
+  data = data.first  # one project per yaml file in projects dir
+  index = project_data.index { |proj| proj['name'] == data['name'] }
+  index = project_data.length  if index.nil?
+  project_data[index] = data
+end
+open(private_projects_yaml_file, 'w') {|outfile| outfile.puts project_data.to_yaml}
```

Comment:
Two more thoughts:
- The Hub's `joiner.rb` plugin uses `HashJoiner` to perform this style of data munging after Jekyll's imported the YAML into `site.data`. In my quest to extract as much of the Hub as possible into the [team_hub gem](https://rubygems.org/gems/team_hub), that may be one of the next bits to extract.
- Still, importing a public version of the data may continue to be useful (so folks outside 18F can build a demo). Hence, this loop might be something to extract into the [hash-joiner gem](https://rubygems.org/gems/hash-joiner) as an executable, as I just did with [filter-yaml-files](https://github.com/18F/hash-joiner/blob/master/bin/filter-yaml-files).